File: solutions/python/01-bencode-string/code/app/main.py
import json
import sys

# import pyparsing - available if you need it!
# import lark - available if you need it!

# Examples:
# 
# - decode_bencode("5:hello") -> "hello"
# - decode_bencode("10:hello12345") -> "hello12345"
def decode_bencode(bencoded_value):
    if bencoded_value[0].isdigit():
        length = int(bencoded_value.split(':')[0])
        return bencoded_value.split(':')[1][:length]
    else:
        raise NotImplementedError("Only strings are supported at the moment")


def main():
    command = sys.argv[1]

    if command == "decode":
        bencoded_value = sys.argv[2]

        print(json.dumps(decode_bencode(bencoded_value)))
    else:
        raise NotImplementedError(f"Unknown command {command}")


if __name__ == "__main__":
    main()

File: solutions/ruby/01-bencode-string/code/app/main.rb
require 'json'

if ARGV.length < 2
  puts "Usage: your_bittorrent.sh <command> <args>"
  exit(1)
end

def decode_bencode(bencoded_value)
  if bencoded_value[0].chr.match?(/\d/)
    first_colon = bencoded_value.index(':')
    return bencoded_value[first_colon+1..-1]
  else
    puts "Only strings are supported at the moment"
    exit(1)
  end
end

command = ARGV[0]

if command == "decode"
    encoded_str = ARGV[1]
    decoded_str = decode_bencode(encoded_str)
    puts JSON.generate(decoded_str)
end
